vllm/attention/backends/mla/utils.py

```diff
@@ -0,0 +1,364 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+import torch
+
+from vllm import _custom_ops as ops
+from vllm import envs
+from vllm.attention.backends.abstract import (AttentionLayer,
+                                              AttentionMetadata,
+                                              MLAAttentionImpl)
+from vllm.distributed import get_tensor_model_parallel_world_size
+from vllm.model_executor.layers.linear import (ColumnParallelLinear,
+                                               RowParallelLinear)
+from vllm.model_executor.layers.rotary_embedding import RotaryEmbedding
+from vllm.vllm_flash_attn import flash_attn_varlen_func
+
+
+@dataclass(kw_only=True)
+class MLAMetadataCommon(AttentionMetadata):
+    # Input positions for rotrary embeddings since for MLA the rotarty
+    # position embeddings are applied inside the attention backend
```

Comment:
```suggestion
    # Input positions for rotrary embeddings since for MLA the rotary
    # position embeddings are applied inside the attention backend
```